unittests/producer_schedule_tests.cpp

```diff
@@ -711,8 +711,9 @@ BOOST_AUTO_TEST_CASE( extra_signatures_test ) try {
       additional_sigs.emplace_back( remote.get_private_key("alice"_n, "bs4").sign(sig_digest) );
 
       // Serialize the augmented additional signatures back into the block extensions.
-      b->block_extensions.clear();
-      emplace_extension(b->block_extensions, additional_sigs_eid, fc::raw::pack( additional_sigs ));
+      std::const_pointer_cast<signed_block>(b)->block_extensions.clear();
```

Comment:
I think we should declare `b` as `mutable_signed_block_ptr` and make a deep copy on line 697:

`b = valid_block;` => `b = std::make_shared<block_ptr>(*valid_block);`

as it is probably not a good idea to modify the block returned by `produce_block`.

An then the cast below is not needed.

Reply:
We could, but the test is about adding an extra signature to an existing block. Seems more in the spirit of the test to manipulate the block itself.

Reply:
I like Greg's suggestion.

The test is taking a copy of a valid block from the `remote` node, corrupting it, and then pushing the corrupted block to the `main` node.

So, I don't see what Greg suggested as going against the spirit of this test.